### PR TITLE
Prepublishing Nudges : create core bottom sheet functionality

### LINKED ISSUE #11602: Prepublishing Nudges : Create the Bottomsheet
Create the Bottomsheet to show post options before publishing:

prepublishing confirmation modal(1)

This task should focus only on displaying the bottom sheet. Options and custom interactions will be covered in the next tasks.

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes #11602
## Findings

The objective of this PR is to create the core structure that's required to support Prepublishing Nudges within the Android app. 

The behavior in Android, will operate and look similarly to the designs below that are done for iOS. 

Screen Shot 2020-04-09 at 1 20 54 PM

## Solution
* To accomplish this the Material Design Bottom Sheet component was utilized. It's already used within the app, so I simply created a BottomSheet dialog that would house the options/actions the user can perform before posting. 

* The nudge is supposed to be triggered when the user is publishing a post for the first time so to make testing possible I modified the `publishNow` call sites within `EditPostActivity`. b5b663bac97fcafed08fd5cb4487af298187d62b

* To make the testing experience more functionality, I decided to create the first fragment that will contain the actions the user performs such as adding tags, changing visibility/scheduling. Currently, this fragment is hardcoded into the `Bottomsheet` layout. In another PR #11634, I will replace this with a `FrameLayout` that will be the container for all fragments.

* The prepublishing actions UI is only being rendered at this point in a barebones way and requires more styling. There are no click listeners as yet as that will also be covered in the navigation PR #11634. 

* The prepublishing actions UI has a state-based approach and the rest of the UI for this screen such as the header, bottom divider and the publish/schedule button will be `UiStates`. These will also be covered later on. 

* The corners of the bottom sheet aren't being rendered with a round border-radius. This is also visible when the create FAB is clicked on the home screen and the bottom sheet is selected. There is a note on the board to [track this](https://github.com/orgs/wordpress-mobile/projects/65#card-36038752). 

* `PrepublishingViewModel.kt` does nothing right now but I had created it as part of the project setup. 

N.B Even though the PR has a lot of files, most of them are just boilerplate view models, adapters, etc. because this is the start of this feature/project 😉 

## Testing

1. Create a new post.
2. Click publish.
3. You should then see the bottom sheet. 

Dark Mode | Light Mode 
--------|-------
   |  

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [x] I have considered adding accessibility improvements for my changes.
- [x] If it's feasible, I have added unit tests. user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jd-alexander** [comment]
Ignore the `Installable Build` failure. It isn't a result of this PR and will be fixed by the platform team 😄

**Review by ashiagr — COMMENTED** [review]
Hi @jd-alexander 👋! I spent some time in going over pre publishing nudges requirements. I liked the way you've started this first PR with great PR description, material components and rtl support in the xmls, and including unit tests. Good job 🙌.

As it mostly includes boilerplate classes and barebones UI, I've just left some minor comments inline :).

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1245,7 +1245,7 @@ private boolean performSecondaryAction() {
                 uploadPost(false);
                 return true;
             case PUBLISH_NOW:
-                showPublishConfirmationDialogAndPublishPost();
+                showPrepublishingNudgeBottomSheetBeforePostUpload();
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Thanks for spotting this 4c67ba4

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsListItemViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class PrepublishingActionsListItemViewHolder(internal val parent: ViewGroup, val uiHelpers: UiHelpers) : ViewHolder(
+        LayoutInflater.from(parent.context).inflate(
+                R.layout.prepublishing_action_list_item, parent, false
+        )
+) {
+    fun bind(uiState: PrepublishingActionItemUiState) {
+        uiState as PrepublishingActionUiState
+        val actionType: TextView = this.itemView.findViewById(R.id.action_type)
+        val actionResult: TextView = this.itemView.findViewById(R.id.action_result)
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsListItemViewHolder.kt]: Resolved in e4fb357

File: WordPress/src/test/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        tagsAction.onActionClicked.invoke(expectedActionType)
+
+        // assert
+        assertThat(viewModel.prepublishingActionType.value).isEqualTo(expectedActionType)
+    }
+
+    private fun getActionUiState(actionType: ActionType): PrepublishingActionUiState {
+        val actions = viewModel.prepublishingActionsUiState.value
+                ?.filterIsInstance(PrepublishingActionUiState::class.java)
+        return actions?.find { it.actionType == actionType }!!
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/test/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModelTest.kt]: Indeed! Thanks! db1dfe0

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.wordpress.android.viewmodel.SingleLiveEvent
+import javax.inject.Inject
+
+class PrepublishingActionsViewModel @Inject constructor() : ViewModel() {
+    private var isStarted = false
+
+    private val _prepublishingActionsUiState = MutableLiveData<List<PrepublishingActionItemUiState>>()
+    val prepublishingActionsUiState: LiveData<List<PrepublishingActionItemUiState>> = _prepublishingActionsUiState
+
+    private val _prepublishingActionType = SingleLiveEvent<ActionType>()
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModel.kt]: Understood. Done in 96302b9

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    fun start() {
+        if (isStarted) return
+        isStarted = true
+        loadActionsUiState()
+    }
+
+    // TODO remove hardcoded Immediately & Public with live data from the EditPostRepository / user changes.
+    private fun loadActionsUiState() {
+        val prepublishingActionsUiStateList = arrayListOf<PrepublishingActionUiState>().apply {
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsViewModel.kt]: Ofcourse we can 🕺 5c093db

File: WordPress/src/main/res/layout/prepublishing_action_list_item.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        android:layout_marginStart="@dimen/margin_extra_large"
+        android:layout_marginTop="@dimen/margin_extra_large"
+        android:textAlignment="viewStart"
+        android:textAppearance="?attr/textAppearanceSubtitle1"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toStartOf="@id/action_result"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/top_divider"
+        app:layout_constraintVertical_chainStyle="spread"
+        tools:text="Yeah" />
```
[the remark this answers is not in the document]
Comment (jd-alexander) [reply WordPress/src/main/res/layout/prepublishing_action_list_item.xml]: WOW. Whoooooooops 😅 Thank you so much for spotting this 😄 1bb6f19

**Review by ashiagr — APPROVED** [review]
Thanks for the changes! LGTM :)

**Pull request merged**